File: backend/split/schemas/bill.py
from pydantic import UUID4, StrictBool, StrictStr

from split.shared.schemas import BaseSchema


class BillUpdateSchema(BaseSchema):
    image: StrictStr


class BillResponseSchema(BaseSchema):
    id: UUID4
    image: StrictStr | None
    running_item_generation: StrictBool
    image_found: StrictBool | None
    borders_detected: StrictBool | None
    generation_successful: StrictBool | None

File: frontend/src/types/api/bill.ts
import type { Nullable } from '@/types/utils';

export interface Bill {
  id: string,
  image: Nullable<string>,
  runningItemGeneration: boolean,
  imageFound: Nullable<boolean>,
  bordersDetected: Nullable<boolean>,
  generationSuccessful: Nullable<boolean>,
}
